packages/cdktf-cli/src/bin/cmds/helper/performance.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import pidtree from "pidtree";
+import pidusage from "pidusage";
+
+async function getCombinedMemoryUsageInBytes(
+  pid = process.pid
+): Promise<number> {
+  const allPids = await pidtree(pid, { root: true });
+  const usages = await pidusage(allPids);
+  return Object.values(usages).reduce(
+    (carry, item) => (item && "memory" in item ? carry + item.memory : carry),
+    0
+  );
+}
+
+export function startPerformanceMonitoring() {
+  const start = Date.now();
+  const memoryMeasurementsInGb: number[] = [];
+
+  const interval = setInterval(async () => {
```

Comment:
nit: Overall, not a big fan of `setInterval`, but I can see how it's convenient right now.

Reply:
Not sure what you would replace it with here, I kinda need an interval :D